From: Alec Leamas <leamas.alec@gmail.com>
Date: Thu, 19 May 2022 21:51:47 +0200
Subject: Run platform check (#2694).

For Debian/Ubuntu/Raspbian packages on arm64 (typically from
backports) issue a warning about non-existing plugins. Also
inform user that Flatpak supports plugins.

Closes: #2694

Forwarded: https://github.com/OpenCPN/OpenCPN/pull/2702
---
 CMakeLists.txt  | 2 ++
 src/chart1.cpp  | 1 +
 src/navutil.cpp | 5 +++++
 3 files changed, 8 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 35a023a..d6bbbd1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -820,6 +820,7 @@ set(
   include/chartdbs.h
   include/chartimg.h
   include/chcanv.h
+  include/check_platform.h
   include/ChInfoWin.h
   include/compass.h
   include/concanv.h
@@ -953,6 +954,7 @@ set(
   src/chartdbs.cpp
   src/chartimg.cpp
   src/chcanv.cpp
+  src/check_platform.cpp
   src/ChInfoWin.cpp
   src/compass.cpp
   src/concanv.cpp
diff --git a/src/chart1.cpp b/src/chart1.cpp
index 826ce90..1a6b201 100644
--- a/src/chart1.cpp
+++ b/src/chart1.cpp
@@ -2941,6 +2941,7 @@ MyFrame::MyFrame(wxFrame *frame, const wxString &title, const wxPoint &pos,
 
   m_resizeTimer.SetOwner(this, RESIZE_TIMER);
   m_recaptureTimer.SetOwner(this, RECAPTURE_TIMER);
+  check_platform(this);
 
 #ifdef __WXOSX__
   // Enable native fullscreen on macOS
diff --git a/src/navutil.cpp b/src/navutil.cpp
index 76ab43d..e8e5b70 100644
--- a/src/navutil.cpp
+++ b/src/navutil.cpp
@@ -480,6 +480,8 @@ static const long long lNaN = 0xfff8000000000000;
 #define NAN (*(double *)&lNaN)
 #endif
 
+bool g_hide_no_plugins_dlg;
+
 // Layer helper function
 
 wxString GetLayerName(int id) {
@@ -575,6 +577,7 @@ int MyConfig::LoadMyConfig() {
   wxDisplaySize(&display_width, &display_height);
 
   //  Set up any defaults not set elsewhere
+  g_hide_no_plugins_dlg = false;
   g_useMUI = true;
   g_TalkerIdText = _T("EC");
   g_maxWPNameLength = 6;
@@ -808,6 +811,7 @@ int MyConfig::LoadMyConfigRaw(bool bAsTemplate) {
     g_memCacheLimit = mem_limit * 1024;  // convert from MBytes to kBytes
 
   Read(_T ( "UseModernUI5" ), &g_useMUI);
+  Read("HideNoPluginsDlg", &g_hide_no_plugins_dlg);
 
   Read(_T( "NCPUCount" ), &g_nCPUCount);
 
@@ -2413,6 +2417,7 @@ void MyConfig::UpdateSettings() {
   Write(_T ( "TrackRotateAt" ), g_track_rotate_time);
   Write(_T ( "TrackRotateTimeType" ), g_track_rotate_time_type);
   Write(_T ( "HighlightTracks" ), g_bHighliteTracks);
+  Write("HideNoPluginsDlg", g_hide_no_plugins_dlg);
 
   Write(_T ( "InitialStackIndex" ), g_restore_stackindex);
   Write(_T ( "InitialdBIndex" ), g_restore_dbindex);
